src/server/pps/server/defaults_test.go

```diff
@@ -246,6 +246,93 @@ func TestAPIServer_CreatePipelineV2_regenerate(t *testing.T) {
 	require.Equal(t, false, req.Autoscaling, "autoscaling is still false")
 }
 
+func TestAPIServer_SetProjectDefaults_regenerate(t *testing.T) {
+	ctx := pctx.TestContext(t)
```

Comment:
WDYT about having a test that lists out test cases as structs of {"field", "ClusterDefault", "ProjectDefault", "SetValue", "regenerate"}, and have expected final value for the field?

Reply:
It’s an intriguing idea. I think one of the key observables is the sequence of steps. Can this be deferred to CORE-1723?

Reply:
yeah totally. A suggestion